text/0000-module-unification.md

```diff
@@ -0,0 +1,837 @@
+the default type represents the default export for that collection.
+
+Here's an illustration of exports from a collection:
+
+```
+src
+  data
+    models
+      author.js <- exports an Author `model`, the default type in the `models` collection
+      comment
+        adapter.js     <- exports a Comment `adapter`
+        model.js       <- exports a Comment `model`
+        serializer.js  <- exports a Comment `serializer`
+```
+
+#### Components
+
+The term "component" has been widely adopted across most front-end frameworks
+to describe a broad swath of UI concerns. Using the same term for the collection
+of template-invocable UI elements will lower the learning curve for developers
+who are new to Ember, while allowing for a useful set of specialized terms to
+flourish to describe particular _types_ of components.
+
+We've already started down the road of component specialization by introducing
+the concept of "routable components". Once we start actually using "routable
+components" in practice, it will become necessary to refer to plain old
+components as something more specific, like "template components". And this
+distinction will probably lead to plain old helpers being referred to as
+"template helpers". Other concepts, such as "Glimmer components" and "template
+component modifiers" will soon be mixed in. We will end up with a multi-faceted
+toolbox available at the template layer which deserves a simple name that
+matches developer expectations. The general term "components" seems a good fit.
+
+### Scope
+
+Developers should understand the available levels of module scope, as well as
+when each is appropriate to use. Scope should be considered when modules are
+generated, and developers should feel free to move modules if they expand or
+contract in scope.
+
+The following levels of scope should be understood:
+
+* Private - private collections should be used when a component or utility
+  function is needed from a single namespace.
+
+* Project - top-level, project-wide collections should be used for modules that
+  are needed throughout a project.
+
+* Local package - namespaced collections can be useful to group a common set of
+  cross-cutting concerns within a project.
+
+* Local engine - a type of local package that encapsulates a set of
+  functionality that benefits from run-time isolation and strict dependency
+  sharing.
+
+### Testing
+
+Unit, integration, and some acceptance tests can now be co-located with their
+associated modules. Co-location should be encouraged because it makes test
+modules easier to locate in the file system, and easier to move if a module's
+scope changes.
+
+Robert Jackson plans to adapt the
+[Grand Testing Unification RFC](https://github.com/emberjs/rfcs/pull/119)
+to illustrate test co-location and to introduce module types for tests.
+
+# Drawbacks
+
+Any change to a pattern as fundamental as file naming will incur some mental
```

Comment:
> On the plus side, this approach will be entirely opt-in for some time and can co-exist with current practices

I understand that, but I'm concerned about when people start to opt-in, and without proper guidance and promotion of the strategies, concepts, etc... I suspect it will result in some negative PR for Ember.

If the path for education here is to point people to the RFC, that is going to fail. There really needs to be some high-level "Ember for beginners"-esque documentation, diagrams, presentations, etc... The amount of effort that Core has put into the RFC should be reflected with the effort they put into the promotion and education of this as well.